### PR TITLE
Always take into account the configured template builder spec in live tests

### PR DESCRIPTION
**Pull request by nacx**
@ivanlomba Can you cherry-pick this commit into your branch, compile it, and try running the live tests for GCE? Run all the tests, not just a single one to make sure all maven configuration for the live profile is properly applied.

### CODE REVIEW

**Comment by ivanlomba** [comment]
Thanks : )

File: compute/src/test/java/org/jclouds/compute/internal/BaseComputeServiceLiveTest.java [hunk 1]
```diff
@@ -524,6 +524,14 @@ protected void checkNodes(Iterable<? extends NodeMetadata> nodes, String group,
       }
    }
 
+   protected TemplateBuilder templateBuilder() {
+      TemplateBuilder templateBuilder =  client.templateBuilder();
```
Comment (demobox) [inline compute/src/test/java/org/jclouds/compute/internal/BaseComputeServiceLiveTest.java]: [formatting] Double space here

**Comment by demobox** [comment]
@nacx @ivanlomba Just to confirm: this commit is only about the *tests*, correct?

**Comment by nacx** [comment]
Yes. The base class for the compute service live tests was ignoring the "template" test property in some tests, so even when configured in the pom.xml or as system property, a default template was used instead of the configured one.

This lead to some tests failing to connect via SSH if the template builder picked a windows machine, etc.
I still want to run the tests in a couple more providers before merging it.

**Comment by demobox** [comment]
> Yes. The base class for the compute service live tests was ignoring the "template" test property in some tests, so even when configured in the pom.xml or as system property, a default template was used instead of the configured one.

Ah, OK - thanks for explaining!

**Comment by nacx** [comment]
I've run the tests in several providers and the fix works fine.
Amended the formatting thing, rebased and merged to master as [449c6d8d](http://git-wip-us.apache.org/repos/asf/jclouds/commit/449c6d8d). Thanks @ivanlomba and @demobox!

**Pull request closed without merging**